handler/interceptors.go

```diff
@@ -372,9 +375,47 @@ func GetSingleValue(md metadata.MD, key string) (value string, err *GrpcError) {
 // NoOpInterceptor is a gRPC interceptor which doesn't do payment checking.
 func NoOpInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo,
 	handler grpc.StreamHandler) error {
+	if config.GetBool(config.AllowedUserFlag){
+		//Just to a validation of the Request here
+
+		context, err := getGrpcContext(ss, info)
+		if err != nil {
+			return err.Err()
+		}
+        if err := checkIfSignerIsAnAllowedUser(context) ;err != nil {
+        	return err
+		}
+
+	}
 	return handler(srv, ss)
 }
+func checkIfSignerIsAnAllowedUser(context *GrpcStreamContext) (err error) {
+	signature, grpcErr := GetBytes(context.MD, PaymentChannelSignatureHeader)
+	if grpcErr != nil {
+		return fmt.Errorf("Metadata %v is not set correctly for curation requests",PaymentChannelSignatureHeader)
+	}
+	blockNumber, grpcErr := GetBigInt(context.MD, CurrentBlockNumberHeader)
```

Comment:
Checking `blockNumber` when blockchain is disabled is impossible so I don't see a reason to include it in header. In any case the first intercepted package can be used for authorization.